azure-pipelines.yml

```diff
@@ -1,44 +1,37 @@
-# Starter pipeline
-# Start with a minimal pipeline that you can customize to build and deploy your code.
-# Add steps that build, run tests, deploy, and more:
 # https://aka.ms/yaml
 
 trigger:
-  - main
+  - "*"
 
 pool:
   vmImage: "ubuntu-latest"
 
 variables:
-  SSH_HOST: "20.215.42.144"
-  SSH_USER: "azureuser"
-  REPO_URL: "https://github.com/maksnowak/FileSharingSystem"
-  WORK_DIR: "/home/azureuser/app"
+  - group: deploy-variables
+  - group: docker-registry-credentials
 
 stages:
-  - stage: Deploy
-    displayName: Deploy Application
-    condition: eq(variables['Build.SourceBranch'], 'refs/heads/main')
+  - stage: BuildAndTest
+    jobs:
+      - template: templates/go-test.yml
+        parameters:
+          appName: "file_transfer"
+          directory: "file-transfer/"
```

Comment:
Builds for file encryption, accounts and frontend will be added in the future, correct?

Reply:
Yes they should be added by team members responsible for them. I provided overall template and it should be modified for the use cases, for example Vue.js probably needs different implementation.